src/main/java/io/getunleash/util/UnleashConfig.java

```diff
@@ -185,6 +186,16 @@ private void enableProxyAuthentication() {
         Authenticator.setDefault(new SystemProxyAuthenticator());
     }
 
+    /**
+     * Fetches client features using this configuration.
+     * If the method does not throw, the configuration was able to fetch features and as such is assumed to be correct.
+     * Use upon initial building of an UnleashClient to simplify debugging and interrupting your application if client is incorrectly configured.
+     * @throws UnleashException Thrown if something went wrong when trying to fetch features. Typical errors are incorrect URLs or apiKeys
+     */
+    public void validate() throws UnleashException {
+        this.getUnleashFeatureFetcherFactory().apply(this).fetchFeatures();
```

Comment:
I'm not sure if this will solve all the issues, maybe we should also assert that the response status is not `UNAVAILABLE`: https://github.com/Unleash/unleash-client-java/blob/e5c3e34d6c47bb1c233ae09db293f17583b33e51/src/main/java/io/getunleash/repository/OkHttpFeatureFetcher.java#L93-L94

Reply:
I agree with @gastonfournier we should make sure to validate that the response is 200. 

I wonder why you would do the validation as part of the configuration object instead of having a similar function on the client itself, then we could make use of the actual response as well as the initial toggle values.